src/Schema/Schema.php

```diff
@@ -65,6 +71,12 @@ class Schema extends AbstractOptionallyNamedObject
 
     protected SchemaConfig $_schemaConfig;
 
+    /**
+     * Indicates whether the schema uses unqualified names for its objects. Once this flag is set to true, it won't be
+     * unset even after the objects with unqualified names have been dropped from the schema.
+     */
+    private bool $usesUnqualifiedNames = false;
```

Comment:
Is this flag write once? Should it be marked as such by using `@readonly`?

Reply:
It's not exactly _write_ once, it can be _flipped_ once from `false` to `true`. `@readonly` might be confusing, if the reader assumes the same contract as the one provided by the `readonly` [modifier](https://www.php.net/manual/en/language.oop5.properties.php#language.oop5.properties.readonly-properties):

> [...] which prevents modification of the property after initialization.

This flag is initialized as `false` and can be modified at any point in time to `true`.